nova/tests/unit/compute/test_api.py

```diff
@@ -2230,6 +2235,29 @@ def test_resize_same_host_and_allowed(self):
     def test_resize_same_host_and_not_allowed(self):
         self._test_resize(same_host=True, allow_same_host=False)
 
+    def test_resize_with_volume_in_valid_state(self):
+        # we pass 2 volumes, so we can test both valid states without calling
+        # _test_resize twice
+        bdms = objects.BlockDeviceMappingList(objects=[
+                objects.BlockDeviceMapping(volume_id=uuids.volume_id),
+                objects.BlockDeviceMapping(volume_id=uuids.volume_id)])
+        fake_volume1 = {'status': "in-use", 'attach_status': 'attached'}
+        fake_volume2 = {'status': "available", 'attach_status': 'detached'}
+        # side_effect returns the values in the iterable on consecutive calls
+        with mock.patch.object(self.compute_api.volume_api, 'get',
+                               side_effect=(fake_volume1, fake_volume2)):
+            self._test_resize(bdms=bdms)
+
+    def test_resize_with_volume_in_invalid_state(self):
+        bdms = objects.BlockDeviceMappingList(objects=[
+                objects.BlockDeviceMapping(volume_id=uuids.volume_id)])
+        fake_volume = {'status': "invalid", 'attach_status': 'invalid',
+                       'id': uuids.volume_id}
```

Comment:
hm ... I don't like that you put states here, the volume couldn't even be in. We could use `@ddt` here to provide a list of states to test against e.g. error/detached, but also in-use/detached and reserved/detached etc.